unmarshal_test.go

```diff
@@ -208,6 +208,16 @@ func TestUnmarshal(t *testing.T) {
 			},
 			expect:      new(Article),
 			expectError: &PartialLinkageError{[]string{"{Type: author, ID: 1}"}},
+		}, {
+			description: "*ArticleRelated empty relationships",
+			given:       articleRelatedNoOmitEmptyBody,
```

Comment:
This tests a document with two relationships for the main resource, one of which is `data: null` and one of which is `data:[]`

Reply:
Probably work a comment